app/kubemci/pkg/gcp/forwardingrule/forwardingrulesyncer_test.go

```diff
@@ -149,21 +253,71 @@ func TestGetLoadBalancerStatus(t *testing.T) {
 	frp := ingresslb.NewFakeLoadBalancers("" /*name*/, nil /*namer*/)
 	namer := utilsnamer.NewNamer("mci1", lbName)
 	frs := NewForwardingRuleSyncer(namer, frp)
-	if err := frs.EnsureHttpForwardingRule(lbName, ipAddr, tpLink, clusters, false /*force*/); err != nil {
-		t.Fatalf("expected no error in ensuring forwarding rule, actual: %v", err)
-	}
-	status, err := frs.GetLoadBalancerStatus(lbName)
-	if err != nil {
-		t.Fatalf("unexpected error in getting status description for load balancer %s", lbName)
-	}
-	// Verify that status description has the expected values set.
-	if status.LoadBalancerName != lbName {
-		t.Errorf("unexpected load balancer name, expected: %s, got: %s", lbName, status.LoadBalancerName)
-	}
-	if !reflect.DeepEqual(status.Clusters, clusters) {
-		t.Errorf("unexpected list of clusters, expected: %v, got: %v", clusters, status.Clusters)
+	testCases := []struct {
+		// Should the http forwarding rule be created.
+		http bool
+		// Should the https forwarding rule be created.
+		https bool
+		// Is get status expected to return an error.
+		shouldErr bool
+	}{
+		{
+			true,
+			false,
+			false,
+		},
+		{
+			false,
+			true,
+			false,
+		},
+		{
+			true,
+			true,
+			false,
+		},
+		{
+			false,
+			false,
+			true,
+		},
 	}
-	if status.IPAddress != ipAddr {
-		t.Errorf("unpexpected ip address, expected: %s, got: %s", status.IPAddress, ipAddr)
+	for i, c := range testCases {
+		glog.Infof("===============testing case: %d=================", i)
+		if c.http {
+			// Ensure http forwarding rule.
+			if err := frs.EnsureHttpForwardingRule(lbName, ipAddr, tpLink, clusters, false /*force*/); err != nil {
+				t.Fatalf("expected no error in ensuring http forwarding rule, actual: %v", err)
+			}
+		}
+		if c.https {
+			// Ensure https forwarding rule.
+			if err := frs.EnsureHttpsForwardingRule(lbName, ipAddr, tpLink, clusters, false /*force*/); err != nil {
+				t.Fatalf("expected no error in ensuring https forwarding rule, actual: %v", err)
+			}
+		}
+		status, err := frs.GetLoadBalancerStatus(lbName)
+		if c.shouldErr != (err != nil) {
+			t.Fatalf("unexpected error in getting status description for load balancer %s, expected err != nil: %v, actual err: %s", lbName, c.shouldErr, err)
+		}
+		if err != nil {
```

Comment:
nit: all the other tests use "c.shouldErr" instead of err

Reply:
done